pkg/apis/v1alpha1/metric_types.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// MetricKind is the kind for metrics.
+	MetricKind = "Metric"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Metric provides metrics configuration.
+type Metric struct {
+	//+k8s:conversion-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata"`
+	// Spec holds spec for metrics.
+	Spec MetricSpec `json:"spec"`
+}
+
+// MetricSpec holds spec for metrics.
+type MetricSpec struct {
+	// Path is a restful service path.
+	Path string `json:"path"`
```

Comment:
Suggestion, if kwokctl is using this configuration, we should add this Path to the Prometheus collection target, so we can open the web of Prometheus and see the chart of the configured metrics.

WDYT?

Reply:
That sounds great as long as we keep it static
If you see it become dynamic (having route like `/nodes/{nodeName}`) in the future, is it okay to opt-out of the prometheus collection targets?

Reply:
Prometheus supports rules for dynamically discovered targets, which will be designed in the future.